spek-ide-plugin/interop-jvm/src/main/kotlin/org/spekframework/ide/sm.kt

```diff
@@ -0,0 +1,112 @@
+        if (result is ExecutionResult.Failure) {
+            val exceptionDetails = getExceptionDetails(result)
+            out("testFailed name='$name' message='${exceptionDetails.first}' details='${exceptionDetails.second}'")
+
+        }
+        out("testFinished name='$name'")
+    }
+
+    override fun testIgnored(test: TestScopeImpl, reason: String?) {
+        val name = test.path.name.toTcSafeString()
+        out("testIgnored name='$name' ignoreComment='${reason ?: "no reason provided"}'")
+        out("testFinished name='$name'")
+    }
+
+    override fun groupExecutionStart(group: GroupScopeImpl) {
+        out("testSuiteStarted name='${group.path.name.toTcSafeString()}'")
+    }
+
+    override fun groupExecutionFinish(group: GroupScopeImpl, result: ExecutionResult) {
+        val name = group.path.name.toTcSafeString()
+        if (result is ExecutionResult.Failure) {
+            val exceptionDetails = getExceptionDetails(result)
+
+            // fake a child test
+            out("testStarted name='$name'")
+            out("testFailed name='$name' message='${exceptionDetails.first}' details='${exceptionDetails.second}'")
+            out("testFinished name='$name'")
+        }
+        out("testSuiteFinished name='$name'")
+    }
+
+    override fun groupIgnored(group: GroupScopeImpl, reason: String?) {
+        val name = group.path.name.toTcSafeString()
+        out("testIgnored name='$name' ignoreComment='${reason ?: "no reason provided"}'")
+        out("testFinished name='$name'")
+    }
+
+    override fun actionExecutionStart(action: ActionScopeImpl) {
+        out("testSuiteStarted name='${action.path.name.toTcSafeString()}'")
+    }
+
+    override fun actionExecutionFinish(action: ActionScopeImpl, result: ExecutionResult) {
+        val name = action.path.name.toTcSafeString()
+        if (result is ExecutionResult.Failure) {
+            val exceptionDetails = getExceptionDetails(result)
+
+            // fake a child test
+            out("testStarted name='$name'")
+            out("testFailed name='$name' message='${exceptionDetails.first}' details='${exceptionDetails.second}'")
+            out("testFinished name='$name'")
+        }
+        out("testSuiteFinished name='$name'")
+    }
+
+    override fun actionIgnored(action: ActionScopeImpl, reason: String?) {
+        val name = action.path.name.toTcSafeString()
+        out("testIgnored name='$name' ignoreComment='${reason ?: "no reason provided"}'")
+        out("testFinished name='$name'")
+    }
+
+    private fun getExceptionDetails(result: ExecutionResult.Failure): Pair<String?, String> {
+        val throwable = result.cause
+        val writer = CharArrayWriter()
+        throwable.printStackTrace(PrintWriter(writer))
+        val details = writer.toString()
+            .toTcSafeString()
+
+        val message = throwable.message?.toTcSafeString()
+
+        return message to details
+    }
+}
+
+private fun String.toTcSafeString(): String {

```

Comment:
`Tc` is TeamCity here?

Reply:
Yep, I'll rename it to `toServiceMessageSafeString()`.